executor.go

```diff
@@ -176,23 +190,39 @@ func (e *Executor) verifySubjectAgainstReferrers(ctx context.Context, task *exec
 		}
 		return nil
 	})
-	if err != nil {
+	if err != nil && err != ErrArtifactPruned {
 		return nil, fmt.Errorf("failed to verify referrers for artifact %s: %w", artifact, err)
 	}
+	if evaluator != nil && evaluator.EvaluateArtifact(ctx, task.artifactDesc.Digest.String()) != nil {
+		return nil, fmt.Errorf("failed to evaluate artifact %s: %w", artifact, err)
+	}		
 	task.subjectReport.ArtifactReports = append(task.subjectReport.ArtifactReports, artifactReports...)
 
+	if err == ErrArtifactPruned {
+		return nil, nil
+	}
 	return newTasks, nil
 }
 
 // verifyArtifact verifies the artifact by all configured verifiers and returns
 // error if any of the verifier fails.
-func (e *Executor) verifyArtifact(ctx context.Context, repo string, subjectDesc, artifact ocispec.Descriptor) ([]*VerificationResult, error) {
+func (e *Executor) verifyArtifact(ctx context.Context, repo string, subjectDesc, artifact ocispec.Descriptor, evaluator Evaluator) ([]*VerificationResult, error) {
 	var verifierReports []*VerificationResult
 
 	for _, verifier := range e.Verifiers {
 		if !verifier.Verifiable(artifact) {
 			continue
 		}
+
+		if evaluator != nil {
+			err := evaluator.Pruned(ctx, subjectDesc.Digest.String(), artifact.Digest.String(), verifier.Name())
+			if err == ErrVerifierPruned {
+				continue
+			} else if err == ErrArtifactPruned {
+				return verifierReports, err
+			}
```

Comment:
What if `error` is neither `ErrVerifierPruned` nor `ErrArtifactPruned`?

Reply:
Good catch, refactored.